sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileApiTests.java

```diff
@@ -637,6 +640,91 @@ public void downloadRetryDefault() {
         assertEquals(bodyStr, DATA.getDefaultText());
     }
 
+    @Test
+    public void downloadToFileWithPartialDownloads() throws Exception {
+        File uploadFile = File.createTempFile(CoreUtils.randomUuid().toString(), ".txt");
+        uploadFile.deleteOnExit();
+        Files.write(uploadFile.toPath(), DATA.getDefaultBytes());
+
+        File outFile = new File(generatePathName() + ".txt");
+        if (outFile.exists()) {
+            assertTrue(outFile.delete());
+        }
+
+        MockPartialResponsePolicy policy = new MockPartialResponsePolicy(5);
+
+        // Create a ShareFileClient for download using the custom pipeline
+        ShareFileClient downloadClient
+            = getFileClient(ENVIRONMENT.getPrimaryAccount().getCredential(), primaryFileClient.getFileUrl(), policy);
+
+        // Upload the test data
+        primaryFileClient.create(DATA.getDefaultDataSize());
+        primaryFileClient.uploadFromFile(uploadFile.toString());
+
+        downloadClient.downloadToFile(outFile.toString());
+
+        byte[] downloadedData = Files.readAllBytes(outFile.toPath());
+        Assertions.assertArrayEquals(DATA.getDefaultBytes(), downloadedData);
+
+        // Assert that we retried the correct number of times (5)
+        assertEquals(0, policy.getTriesRemaining());
+
+        // Assert that the range headers that were retried match what was returned from MockPartialResponsePolicy
+        List<int[]> actualRanges = policy.getRangesList();
+        for (int i = 0; i < actualRanges.size(); i++) {
+            int[] range = actualRanges.get(i);
+            assertEquals(i, range[0]);
+            assertEquals(DATA.getDefaultDataSize() - 1, range[1]);
+        }
+
+        // Clean up
+        Files.deleteIfExists(outFile.toPath());
+        Files.deleteIfExists(uploadFile.toPath());
```

Comment:
`deleteOnExit()` will delete once the JVM terminates. By calling `Files.deleteIfExists()`, it will ensure that the files get deleted immediately after this test finishes executing instead of waiting for the JVM to terminate. I can go ahead and add a `deleteOnExit()` for the outFile as well.

Reply:
That sounds good